DOC: first pass at switching to pydata theme

tacaswell opened the pull request:
## PR Summary

~Still super busted.~

Builds clean and looks mostly OK!

jklymak commented:
Did you ever get this to build?

tacaswell commented:
It builds locally, I forgot to add the extra dependency to the requirements :facepalm: .

tacaswell commented:
It at least builds!

jklymak commented:
Nice... https://55478-1385122-gh.circle-artifacts.com/0/doc/build/html/index.html  Obviously the front page is a mess, but we are replacing anyway.  The rest definitely shows promise...

timhoffm commented:
Looks quite good. Some open style issues:

- [ ] The text boxes in https://55478-1385122-gh.circle-artifacts.com/0/doc/build/html/devel/triage.html
- [ ] The Quick Buttons in https://55478-1385122-gh.circle-artifacts.com/0/doc/build/html/devel/index.html are not styled anymore (this used custom CSS we introduced)
- [ ] TOC should be removed from https://55478-1385122-gh.circle-artifacts.com/0/doc/build/html/devel/documenting_mpl.html
- [ ] Some pictures in https://55478-1385122-gh.circle-artifacts.com/0/doc/build/html/thirdpartypackages/index.html are too wide and thus overflow.
- [ ] Gallery only fits 3 thumbnails per row (not sure if that's a problem)
- [ ] Gallery thumbnails have a red marker between image and title. https://55478-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/index.html

tacaswell commented:
Made more progress!  Figured out that we were fully shadowing core style sheet which is why so many things looked broken.

I have addressed the issues from @timhoffm above.  It bulids clean locally,

tacaswell commented:
If this is clean and good enough, can we merge this quickly and iterate?

jklymak reviewed (approved):
I'm fine w/ this going in.  The front page looks kind of bad, but it is its own thing now, right?

tacaswell commented:
>  The front page looks kind of bad, but it is its own thing now, right?

There needs to be a bit more work on that, but yes.

QuLogic commented on doc/conf.py:
```diff
@@ -266,42 +269,73 @@ def _check_dependencies():
-# html_logo = 'logo.png'
+html_logo = "_static/logo2.svg"
+html_theme_options = {
+    "logo_link": "/stable/index.html",
```
Is that what we want on versioned pages?

tacaswell replied:
Probably not.  This also makes the locally built version of the docs not self-contained (so does the third party page link, but that includes the full URL so it atleast works (assuming you have network).  I'll change this.

QuLogic commented on doc/conf.py:
```diff
@@ -24,16 +24,16 @@
 # If your extensions are in another directory, add it here. If the directory
 # is relative to the documentation root, use os.path.abspath to make it
 # absolute, like shown here.
-sys.path.append(os.path.abspath('.'))
```
There's a lot of unnecessary churn here; you really should configure `black`/`pre-commit`/whatever to only work on the diff you actually touched.

tacaswell replied:
I'll strip out the churn.

QuLogic commented on doc/users/index.rst:
```diff
@@ -12,11 +12,11 @@ User's Guide
-    history.rst
     license.rst
     ../citing.rst
+    ../resources/index.rst
```
Should we move this directory?

tacaswell replied:
maybe, but that would change the URLs.

tacaswell commented:
At this point, I do not think my struggles to get this working are of interest to anyone so squashed.

timhoffm reviewed (approved):
I'm fine with this going in as a first version. Still needs a bit of polishing here and there, bit iIt's simpler to do additional refinement incrementally.

tacaswell commented:
Given the two approvals I pushed the button on this!

jklymak commented:
We thought to try reverting this to see if the doc build would actually speed up, but Github cannot revert it "the content may have changed since it was merged." so if we try to revert, it'll be manual.

timhoffm commented:
Um, can‘t you simply brach from an old commit? It‘s not exactly the same repo state and docs then, but I suppose that shouldn’t make the difference.
